CorsixTH/Lua/world.lua

```diff
@@ -1874,23 +1879,27 @@ function World:findRoomNear(humanoid, room_type_id, distance, mode)
   if not distance then
     distance = 2^30
   end
-  for _, r in pairs(self.rooms) do repeat
-    if r.built and (not room_type_id or r.room_info.id == room_type_id) and r.is_active then
+
+  for _, r in pairs(self.rooms) do
+    if r.built and (not room_type_id or r.room_info.id == room_type_id) and
```

Comment:
I don't know how often this room type check is done in the code, but would creating a `Room::isOfType(room_type)` wrapper function be useful?

Reply:
I assume you mean moving that bracketed query into the `room` code? I mean maybe... though as per my other comment, is `not room_type_id` some weird legacy safeguard?

Reply:
`if room,room_info.id == x` is used ~40 times. Not sure I want to consolidate it right now.

Reply:
Fair enough.

The idea of such wrapper functions is that `r:isRoomOfType(room_type_id)` is more clear wrt its intention (even more if the `_id`suffix gets dropped). It also raises the abstraction level a bit. The spot where a room is storing its type is then more an internal `Room` thing instead of the 40 places now where the knowledge is sprinkled through the code.